subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/use/NonDeclarativePluginUseIntegrationSpec.groovy

```diff
@@ -309,7 +309,7 @@ class NonDeclarativePluginUseIntegrationSpec extends AbstractIntegrationSpec {
 
         and:
         failure.assertHasDescription("An exception occurred applying plugin request [id: 'org.myplugin', version: '1.0']")
-        failure.assertHasCause("throwing plugin")
+        failure.assertHasCause("java.lang.Exception: throwing plugin")
```

Comment:
It seems to me that we need to do some unwrapping. Clearly this is a less readable message than before. The build operation executor could throw a special kind of exception that Gradle knows to unwrap before presenting the cause to the user.